iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/AffiliationConverterTest.java

```diff
@@ -0,0 +1,225 @@
+        
+        ExtractedDocumentMetadata document = new ExtractedDocumentMetadata();
+        document.setId("XYZ");
+        
+        Affiliation aff1 = createAffiliation();
+        aff1.setOrganization(null);
+        aff1.setCountryName(null);
+        aff1.setCountryCode(null);
+        
+        when(organizationNameNormalizer.normalize("")).thenReturn("X");
+        when(countryNameNormalizer.normalize("")).thenReturn("X");
+        when(countryCodeNormalizer.normalize("")).thenReturn("X");
+        
+        List<Affiliation> affiliations = Lists.newArrayList(aff1);
+        document.setAffiliations(affiliations);
+
+        
+        // execute
+        
+        List<AffMatchAffiliation> affMatchAffiliations = converter.convert(document);
+        
+        
+        // assert
+        
+        assertNotNull(affMatchAffiliations);
+        assertAffiliation(affMatchAffiliations.get(0), "XYZ", 1, "X", "X", "X");
+        
+    }
+
+    
+    
+    @Test
+    public void convert() {
+        
+        // given
+        
+        ExtractedDocumentMetadata document = new ExtractedDocumentMetadata();
+        document.setId("XYZ");
+        
+        Affiliation aff1 = createAffiliation();
+        Affiliation aff2 = createAffiliation();
+        Affiliation aff3 = createAffiliation();
+        
+        when(organizationNameNormalizer.normalize(aff1.getOrganization().toString())).thenReturn("abc inst name");
+        when(organizationNameNormalizer.normalize(aff2.getOrganization().toString())).thenReturn("def inst name");
+        when(organizationNameNormalizer.normalize(aff3.getOrganization().toString())).thenReturn("ghi inst name");
+        
+        when(countryNameNormalizer.normalize(aff1.getCountryName().toString())).thenReturn("Poland");
+        when(countryNameNormalizer.normalize(aff2.getCountryName().toString())).thenReturn("Germany");
+        when(countryNameNormalizer.normalize(aff3.getCountryName().toString())).thenReturn("Slovakia");
+        
+        when(countryCodeNormalizer.normalize(aff1.getCountryCode().toString())).thenReturn("pl");
+        when(countryCodeNormalizer.normalize(aff2.getCountryCode().toString())).thenReturn("de");
+        when(countryCodeNormalizer.normalize(aff3.getCountryCode().toString())).thenReturn("sl");
+        
+        List<Affiliation> affiliations = Lists.newArrayList(aff1, aff2, aff3);
+        document.setAffiliations(affiliations);
+
+        
+        // execute
+        
+        List<AffMatchAffiliation> affMatchAffiliations = converter.convert(document);
+        
+        
+        // assert
+        
+        assertNotNull(affMatchAffiliations);
+        assertAffiliation(affMatchAffiliations.get(0), "XYZ", 1, "abc inst name", "Poland", "pl");
+        assertAffiliation(affMatchAffiliations.get(1), "XYZ", 2, "def inst name", "Germany", "de");
+        assertAffiliation(affMatchAffiliations.get(2), "XYZ", 3, "ghi inst name", "Slovakia", "sl");
+        
+    }
+
```

Comment:
If two generated affiliations will get the same country code randomly - won't the test fail?